// @uiw/react-github-corners/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = githubCorners;
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireDefault(require("react"));
require("@uiw/github-corners");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["size", "fixed", "bottom", "zIndex", "className", "style", "bgColor", "color", "position"];
function githubCorners(props) {
  var _props$size = props.size,
    size = _props$size === void 0 ? 80 : _props$size,
    _props$fixed = props.fixed,
    fixed = _props$fixed === void 0 ? false : _props$fixed,
    bottom = props.bottom,
    zIndex = props.zIndex,
    className = props.className,
    style = props.style,
    _props$bgColor = props.bgColor,
    bgColor = _props$bgColor === void 0 ? '#151513' : _props$bgColor,
    _props$color = props.color,
    color = _props$color === void 0 ? '#fff' : _props$color,
    _props$position = props.position,
    position = _props$position === void 0 ? 'right' : _props$position,
    otherProps = (0, _objectWithoutProperties2.default)(props, _excluded);
  var styl = position === 'left' ? {
    left: 0,
    right: 'initial',
    transform: 'scale(-1, 1)'
  } : {
    right: 0,
    left: 'initial',
    transform: 'scale(1, 1)'
  };
  if (bottom) {
    styl.bottom = 0;
    styl.top = 'initial';
    styl.transform = position === 'left' ? 'scale(-1, -1)' : 'scale(1, -1)';
  } else {
    styl.bottom = 'initial';
    styl.top = 0;
  }
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("github-corners", (0, _objectSpread2.default)((0, _objectSpread2.default)({
    target: "__blank",
    width: size,
    height: size,
    href: props.href,
    position: fixed ? 'fixed' : 'absolute',
    "z-index": zIndex,
    style: style,
    fill: bgColor,
    color: color
  }, styl), otherProps));
}
module.exports = exports.default;
