// @uiw/react-markdown-preview-example/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.SupVersion = void 0;
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));
var _react = require("react");
require("@wcj/dark-mode");
var _styledComponents = require("styled-components");
var _reactBackToTop = _interopRequireDefault(require("@uiw/react-back-to-top"));
var _Github = require("./Github");
var _Corners = require("./Corners");
var _Example = require("./Example");
var _NavMenu = require("./NavMenu");
var _store = require("./store");
var _Markdown = _interopRequireDefault(require("./Markdown"));
var _Logo = require("./Logo");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["version", "title", "description", "source", "logo", "components", "data", "markdownProps", "exampleProps", "className", "children", "disableCorners", "disableDarkMode", "disableHeader", "disableBackToUp"];
var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5;
var ExampleWrapper = _styledComponents.styled.div(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2["default"])(["\n  max-width: 56rem;\n  margin: 0 auto;\n  padding: 2.3rem 3rem;\n  display: flex;\n  justify-content: center;\n"])));
var Wrappper = _styledComponents.styled.div(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2["default"])(["\n  padding-bottom: 12rem;\n"])));
var Header = _styledComponents.styled.header(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2["default"])(["\n  padding: 9rem 0 2rem 0;\n  text-align: center;\n  h1 {\n    font-weight: 900;\n    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif,\n      'Apple Color Emoji', 'Segoe UI Emoji';\n  }\n"])));
var SupVersion = exports.SupVersion = _styledComponents.styled.sup(_templateObject4 || (_templateObject4 = (0, _taggedTemplateLiteral2["default"])(["\n  font-weight: 200;\n  font-size: 0.78rem;\n  margin-left: 0.5em;\n  margin-top: -0.3em;\n  position: absolute;\n  white-space: nowrap;\n"])));
var Description = _styledComponents.styled.p(_templateObject5 || (_templateObject5 = (0, _taggedTemplateLiteral2["default"])(["\n  max-width: 460px;\n  margin: 0 auto;\n  color: var(--color-fg-subtle, #6e7781);\n"])));
var InternalMarkdownPreviewExample = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  var version = props.version,
    title = props.title,
    description = props.description,
    source = props.source,
    _props$logo = props.logo,
    logo = _props$logo === void 0 ? _Logo.Logo : _props$logo,
    components = props.components,
    data = props.data,
    markdownProps = props.markdownProps,
    exampleProps = props.exampleProps,
    _props$className = props.className,
    className = _props$className === void 0 ? '' : _props$className,
    children = props.children,
    _props$disableCorners = props.disableCorners,
    disableCorners = _props$disableCorners === void 0 ? false : _props$disableCorners,
    _props$disableDarkMod = props.disableDarkMode,
    disableDarkMode = _props$disableDarkMod === void 0 ? false : _props$disableDarkMod,
    _props$disableHeader = props.disableHeader,
    disableHeader = _props$disableHeader === void 0 ? false : _props$disableHeader,
    _props$disableBackToU = props.disableBackToUp,
    disableBackToUp = _props$disableBackToU === void 0 ? false : _props$disableBackToU,
    reset = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var store = (0, _store.useStores)();
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(Wrappper, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    className: "wmde-markdown-var ".concat(className)
  }, reset), {}, {
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_NavMenu.NavMenuView, {
      version: version,
      logo: logo,
      disableDarkMode: disableDarkMode,
      disableCorners: disableCorners
    }), !disableHeader && /*#__PURE__*/(0, _jsxRuntime.jsxs)(Header, {
      children: [logo, title && /*#__PURE__*/(0, _jsxRuntime.jsxs)("h1", {
        children: [title, version && /*#__PURE__*/(0, _jsxRuntime.jsx)(SupVersion, {
          children: version
        })]
      }), description && /*#__PURE__*/(0, _jsxRuntime.jsx)(Description, {
        children: description
      })]
    }), store.example && /*#__PURE__*/(0, _jsxRuntime.jsx)(ExampleWrapper, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, exampleProps), {}, {
      children: store.example
    })), /*#__PURE__*/(0, _jsxRuntime.jsx)(_Markdown["default"], (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, markdownProps), {}, {
      source: source,
      data: {
        data: data,
        components: components,
        source: source
      }
    })), children, !disableBackToUp && /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactBackToTop["default"], {
      children: "Top"
    })]
  }));
});
var MarkdownPreviewExample = InternalMarkdownPreviewExample;
MarkdownPreviewExample.Github = _Github.Github;
MarkdownPreviewExample.Corners = _Corners.Corners;
MarkdownPreviewExample.Example = _Example.Example;
MarkdownPreviewExample.NavMenu = _NavMenu.NavMenu;
var _default = exports["default"] = MarkdownPreviewExample;
