// @uiw/react-markdown-preview-example/cjs/Logo.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Logo = void 0;
var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));
var _styledComponents = require("styled-components");
var _jsxRuntime = require("react/jsx-runtime");
var _templateObject, _templateObject2;
var rotate = (0, _styledComponents.keyframes)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2["default"])(["\n  from {\n    transform: rotate(0deg);\n  }\n\n  to {\n    transform: rotate(360deg);\n  }\n"])));
var SVG = _styledComponents.styled.svg(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2["default"])(["\n  color: var(--rmpe-logo-color, #61dafb);\n  height: 8rem;\n  margin-bottom: 3rem;\n  pointer-events: none;\n  animation: ", " infinite 20s linear;\n"])), rotate);
var Logo = exports.Logo = /*#__PURE__*/(0, _jsxRuntime.jsxs)(SVG, {
  width: "100%",
  height: "100%",
  viewBox: "-10.5 -9.45 21 18.9",
  fill: "none",
  children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("circle", {
    cx: "0",
    cy: "0",
    r: "2",
    fill: "currentColor"
  }), /*#__PURE__*/(0, _jsxRuntime.jsxs)("g", {
    stroke: "currentColor",
    strokeWidth: "1",
    fill: "none",
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("ellipse", {
      rx: "10",
      ry: "4.5"
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)("ellipse", {
      rx: "10",
      ry: "4.5",
      transform: "rotate(60)"
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)("ellipse", {
      rx: "10",
      ry: "4.5",
      transform: "rotate(120)"
    })]
  })]
});
